src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -815,14 +861,18 @@ ExecuteTaskListExtended(RowModifyLevel modLevel, List *taskList,
 static DistributedExecution *
 CreateDistributedExecution(RowModifyLevel modLevel, List *taskList, bool hasReturning,
 						   ParamListInfo paramListInfo, TupleDesc tupleDescriptor,
-						   Tuplestorestate *tupleStore, int targetPoolSize)
+						   Tuplestorestate *tupleStore, int targetPoolSize,
+						   bool errorOnAnyFailure,
+						   TransactionBlocksUsage useRemoteTransactionBlocks)
 {
 	DistributedExecution *execution =
 		(DistributedExecution *) palloc0(sizeof(DistributedExecution));
 
 	execution->modLevel = modLevel;
 	execution->tasksToExecute = taskList;
 	execution->hasReturning = hasReturning;
+	execution->errorOnAnyFailure = errorOnAnyFailure;
+	execution->useRemoteTransactionBlocks = useRemoteTransactionBlocks;
```

Comment:
At this point we can already determine what we want this value to be, right?. However, we might change this in InitialiseCoordinatedTransactionForTaskList, I feel like we should assign the last value (which will be used during the execution) here. Since there is only one method to create an execution, this will be the single source of truth. What do you think?

Reply:
Not sure I follow, this is called after InitialiseCoordinatedTransactionForTaskList.

Reply:
my bad, I thought it was called afterwards